### ISSUE TITLE
Possible bug in EntityWorld.Clear() ??

### ISSUE THREAD

**Issue by androidedev** [thelinuxlich/artemis_CSharp#86]
i don't know if this is a bug, is possible that is working as expected but due to lack of a more general documentation i'm missunderstanding it.

The problem is when you call EntityWorld.Clear() after some Entities had been added and processed artemis crash. 

To check it simply clone the repo put the line entityWorld.Clear(); before close brace of TestSimpleSystem() in TestGeneral.cs and run the test. 

The code will look like this:
...
            const float Expected2 = 100.0f;
            Assert.AreEqual(Expected2, entity2.GetComponent<TestHealthComponent>().Points);
            Assert.AreEqual(Expected2, entity2.GetComponent<TestPowerComponent>().Power);
            entityWorld.Clear(); // >>> NEW LINE
        }

Tracing it the problem seems to be this call secuence :

1.EntityWorld.Clear() > delete all entities in EntityManager.ActiveEntities  > ok
2.EntityWorld.Clear() > calls EntityWorld.Update() > ok
3.EntityWorld.Update() > delete all marked entities in deleted bag and removes it from EntityManager > ok
4.EntityWorld.Update() > calls SystemManager.Update() > ok
5.SystemManager.Update() > calls SystemManager.Process for updatelayers > ok
6.SystemManager.Process() > calls ProcessBagSynchronous() > ok
7.ProcessBagSynchronous() > calls each one of entity systems Process() > ok
8.EntitySystem.Process() > calls ProcessEntities with this.actives > error, actives has entities despite of beign cleared before > raise exception

is and error or i'm misusing EntityWorld.Clear() and is not supposed to be called in this way.

thanks in advance.

**Comment by Maximusya** [comment]
First of all, that `TestSimpleSystem()` test method uses a sample system that is not so simple - it's is rather an antipattern:
```csharp
public class TestNormalEntityProcessingSystem1 : EntityProcessingSystem
{
	/// <summary>Initializes a new instance of the <see cref="TestNormalEntityProcessingSystem1" /> class.</summary>
	public TestNormalEntityProcessingSystem1()
		: base(Aspect.Exclude(typeof(TestPowerComponent)))
	{
	}

	/// <summary>The process.</summary>
	/// <param name="entity">The entity.</param>
	public override void Process(Entity entity)
	{
		entity.GetComponent<TestHealthComponent>().AddDamage(10);
	}
}
```

See - nothing in the system's aspect declares that `TestHealthComponent` is required, yet, the Process-code blindly gets the `TestHealthComponent`. NRE is almost welcomed here)

Secondly, `EntityWorld.Clear()` method is not covered by any test, it is not documented - actually it is almost never used in the software I've seen! Hence a surprise outcome of your contra-test. The buggy "clearing" causes the deleted entity to actually be _added_ to the system in your test!

So to sum up: both the `TestNormalEntityProcessingSystem1` and `EntityWorld.Clear()` are buggy.

**Comment by androidedev** [comment]
I see, thank you so much Maximusya. 

Before I followed the development of Artemis often, but for some time i stop to follow, now I'm back to follow him and when i see that there were only five issues thought everything else was already tested and stable. 

By the way who is now responsible for maintaining Artemis and merge to master branch? still thelinuxlich?

**Comment by tpastor** [comment]
Clear was not supposed to exist or be public ...
The tests were very plain and only built to test some very specific bug. 
We dont have time to maintain this api anymore. We are looking for people that want to perform this task. If someone is interest, send us a message !
